pandas/tests/test_multilevel.py

```diff
@@ -1609,12 +1607,12 @@ def test_constructor_with_tz(self):
         )
 
         result = MultiIndex.from_arrays([index, columns])
-        tm.assert_index_equal(result.levels[0], index)
-        tm.assert_index_equal(result.levels[1], columns)
+        tm.assert_index_equal(result.levels[0], index, check_names=False)
```

Comment:
I now find these tests very confusing that we lose the names on the levels themselves. (I know that's the point of this PR).

Reply:
maybe add a .set_names(index.name) (for example) and remove the check_names arg (so its the default of True)

Reply:
I've made changes so we avoid ``check_names`` is not used anymore (so is implicitly ``True``.